packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -153,6 +153,15 @@ describe('CozyClient initialization', () => {
     expect(client.stackClient.token.token).toBe(token)
   })
 
+  it('can be instantiated from dataset injected by the Stack in DOM', async () => {
+      const options = { cozyDomain: 'cozy.tools', cozyToken: 'abc123' }
+      const globalQuerySelectorBefore = document.querySelector
+      document.querySelector = jest.fn().mockReturnValue({ dataset: options })
```

Comment:
I think you want to use jest.spyOn to temporarily replace document.querySelector.

```
beforeEach(() => {
  jest.spyOn(document, 'querySelector').mockReturnValue({ dataset: options })
})

afterEach(() => {
  document.querySelector.mockRestore()
})
```